inferno-ml/app/Main.hs

```diff
@@ -14,13 +14,15 @@ import System.Environment (getArgs)
 
 main :: IO ()
 main = do
-  file <- head <$> getArgs
-  src <- Text.readFile file
-  Interpreter {evalExpr, defaultEnv, parseAndInferTypeReps} <-
-    mkInferno @_ @(MlValue ()) mlPrelude customTypes
-  case parseAndInferTypeReps src of
-    Left err -> print err
-    Right ast ->
-      evalExpr defaultEnv Map.empty ast >>= \case
+  getArgs >>= \case
```

Comment:
I tested this executable fairly early on so changed it to avoid `head` (which emits a warning on GHC 9.10). However, there are unfortunately several other places in the codebase where `head` is used. I've decided to leave that for another day, as it involves a fairly large refactor (either that or disabling the partial warning, which I think is a good thing and should be left enabled)